File: 1544.make-the-string-great/solution.py
class Solution:
    def makeGood(self, s: str) -> str:
        stack = []
        for char in s:
            isLower = char.islower()
            lwChar = char.lower()
            upChar = char.upper()

            if (stack):
                lastIsUpper = stack[-1].isupper()
                if ((isLower and lastIsUpper and upChar == stack[-1]) or
                   (not isLower and not lastIsUpper and lwChar == stack[-1])):
                    stack.pop()
                    continue
            stack.append(char)

        return "".join(stack)

File: 1544.make-the-string-great/solution.kt
class Solution {
    fun makeGood(s: String): String {
        val stack = Stack<Char>()
        for (char in s) {
            if (stack.size > 0) {
                val isLower = char.isLowerCase()
                val lastChar = stack.peek()
                val isLastLower = lastChar.isLowerCase()

                val lwChar = char.toLowerCase()
                val upChar = char.toUpperCase()

                if ((isLastLower && !isLower && lwChar == lastChar) ||
                    (!isLastLower && isLower && upChar == lastChar)) {
                    stack.pop()
                    continue
                }
            }
            stack.push(char)
        }
        var result = ""
        while (stack.size > 0) {
            result = "${stack.pop()}${result}"
        }
        return result
    }
}
